File: security_cam/test/detection_region_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:security_cam/core/models.dart';

void main() {
  const rect = DetectionRegion(
    id: 'r1',
    shape: 'rect',
    label: 'doorway',
    points: [0.1, 0.2, 0.5, 0.8],
  );

  test('rect JSON round-trips', () {
    final back = DetectionRegion.fromJson(rect.toJson());
    expect(back.id, 'r1');
    expect(back.shape, 'rect');
    expect(back.label, 'doorway');
    expect(back.points, [0.1, 0.2, 0.5, 0.8]);
  });

  test('poly JSON round-trips', () {
    const poly = DetectionRegion(
      id: 'p1',
      shape: 'poly',
      label: 'driveway',
      points: [0.5, 0.2, 0.8, 0.3, 0.9, 0.6, 0.4, 0.8],
    );
    final back = DetectionRegion.fromJson(poly.toJson());
    expect(back.shape, 'poly');
    expect(back.label, 'driveway');
    expect(back.points, [0.5, 0.2, 0.8, 0.3, 0.9, 0.6, 0.4, 0.8]);
  });
}

File: level2/android/app/src/test/kotlin/io/securitycam/level2/detection/DetectionRegionTest.kt
package io.securitycam.level2.detection

import org.junit.Assert.assertEquals
import org.junit.Test

/** Port of `test/detection_region_test.dart`. */
class DetectionRegionTest {

    private val rect = DetectionRegion(
        id = "r1",
        shape = "rect",
        label = "doorway",
        points = listOf(0.1, 0.2, 0.5, 0.8),
    )

    @Test
    fun rectJsonRoundTrips() {
        val back = DetectionRegion.fromJson(rect.toJson())
        assertEquals("r1", back.id)
        assertEquals("rect", back.shape)
        assertEquals("doorway", back.label)
        assertEquals(listOf(0.1, 0.2, 0.5, 0.8), back.points)
    }

    @Test
    fun polyJsonRoundTrips() {
        val poly = DetectionRegion(
            id = "p1",
            shape = "poly",
            label = "driveway",
            points = listOf(0.5, 0.2, 0.8, 0.3, 0.9, 0.6, 0.4, 0.8),
        )
        val back = DetectionRegion.fromJson(poly.toJson())
        assertEquals("poly", back.shape)
        assertEquals("driveway", back.label)
        assertEquals(listOf(0.5, 0.2, 0.8, 0.3, 0.9, 0.6, 0.4, 0.8), back.points)
    }
}